src/routers/user/routeValidator.ts

```diff
@@ -12,15 +12,17 @@ const routeValidator = (route: string) => {
         check('paid').optional().isBoolean(),
         check('picture').optional().isString().trim(),
       ];
-    case '/membership-check':
+    case '/membership':
+      return [check('membershipStatus').notEmpty(), query('id').notEmpty()];
+    case '/membership/check':
       return [
         oneOf(
           [
-            check('emailOrWatIAMUserId')
+            query('emailOrWatIAMUserId')
```

Comment:
`check` will validate from all of the query, body, params and a few other places so it doesn't need to be replaced here unless the goal is to be more specific. Then we should replace `check` with `body` elsewhere.

Actually I think we definitely should replace all the `check`s with the more specific `body` or whichever necessary since I bet validation would incorrectly pass if the parameters to request are sent in the wrong place (expected in body but sent in query).

Reply:
I agree, we can add that as a refactor

Reply:
Created a card for this refactor in the BL : https://trello.com/c/LW8TuoI7